src/Streamly/Prelude.hs

```diff
@@ -677,8 +676,11 @@ fromIndices = fromStreamD . D.fromIndices
 --
 -- @since 0.6.0
 {-# INLINE fromIndicesM #-}
-fromIndicesM :: (IsStream t, Monad m) => (Int -> m a) -> t m a
-fromIndicesM = fromStreamD . D.fromIndicesM
+fromIndicesM :: (IsStream t, MonadAsync m) => (Int -> m a) -> t m a
```

Comment:
1) This will have bad performance in the SerialT case. We need to use rewrite rules to rewrite this to `D.fromIndicesM` in serial case. See the implementation of `unfoldrM` for an example.

2) Move this implementation to StreamK module and use `K.fromIndicesM` here.